### PR TITLE
v6 TextInput, TagInput, and Textarea all in new theme paradigm.

### PR DESCRIPTION
**Pull request by akleiner2**
## Overview
Backwards-compat input, taginput, and textarea in the new theming paradigm 👍 

## Testing

- [ ] Updated Typescript types and/or component PropTypes
- [ ] Added / modified component docs
- [ ] Added / modified Storybook stories

### CODE REVIEW

File: src/text-input/src/TextInput.js [hunk 1]
```diff
@@ -2,7 +2,26 @@ import React, { forwardRef, memo } from 'react'
...
-import useInputAppearance from '../../theme/src/hooks/useInputAppearance'
+import useStyleConfig from '../../hooks/use-style-config'
+
+const pseudoSelectors = {
+  _focus: '&:focus',
+  _disabled: '&:disabled',
+  _invalid: '&[aria-invalid="true"]',
+  _placeholder: '&::placeholder',
+  _placeholderHover: '&:hover::placeholder',
+  _placeholderFocus: '&:focus::placeholder'
```
Comment (mshwery) [inline src/text-input/src/TextInput.js:13]: I'm still not a huge fan of placeholder hover/focus states being different :)

File: src/themes/classic/components/index.js [hunk 2]
```diff
@@ -4,12 +4,16 @@ import Code from './code'
...
 
 export default {
   Alert,
   Button,
   Code,
   Pane,
   Select,
-  Tooltip
+  Tooltip,
+  Input,
```
Comment (mshwery) [inline src/themes/classic/components/index.js:17]: Mind alphabetizing? (should really be automatic formatting imo)

File: src/textarea/src/Textarea.js [hunk 3]
```diff
@@ -26,7 +41,12 @@ const Textarea = memo(
       ...restProps
     } = props
 
-    const { className: themedClassName, boxProps } = useInputAppearance()
+    const { className: themedClassName, ...boxProps } = useStyleConfig(
+      'Input',
```
Comment (mshwery) [inline src/textarea/src/Textarea.js:45]: We probably need to call out that these don't match the component names in the theming docs.

Comment (mshwery) [reply src/textarea/src/Textarea.js:45]: Or in the theme you could split them up and import the same file for both exports `TextInput` and `Textarea`?

File: src/themes/classic/components/input.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  fontFamily: 'fontFamilies.ui',
+  lineHeight: '12px',
+  border: '1px solid transparent'
+}
+
+const appearances = {
+  default: {
+    backgroundColor: 'white',
+    boxShadow: theme =>
+      `inset 0 0 0 1px ${theme.scales.neutral.N5A}, inset 0 1px 2px ${theme.scales.neutral.N4A}`,
```
Comment (mshwery) [inline src/themes/classic/components/input.js:12]: Is this in the `shadows`?

**Review by mshwery — APPROVED** [review]
🔥

File: src/themes/classic/components/index.js (hunk 2 above)
Comment (akleiner2) [reply src/themes/classic/components/index.js:17]: i'll follow up with something that'll enforce this more broadly across the eg codebase 👍

File: src/textarea/src/Textarea.js (hunk 3 above)
Comment (akleiner2) [reply src/textarea/src/Textarea.js:45]: Yeah, I definitely want to break these apart.

**Pull request merged**